src/rpc/misc.cpp

```diff
@@ -421,6 +422,42 @@ UniValue setmocktime(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+UniValue mnauth(const JSONRPCRequest& request)
+{
+    if (request.fHelp || (request.params.size() != 3))
+        throw std::runtime_error(
+            "mnauth nodeId \"proTxHash\" \"publicKey\"\n"
+            "\nFake a MNAUTH for a connected node in regression test (-regtest only).\n"
```

Comment:
nit: How about smth like
```suggestion
            "\nOverride MNAUTH processing results for the specified node with a user provided data (-regtest only).\n"
```
?